department-of-veterans-affairs/vets-website
# [Bugfix] SSO auto sign-in bug

## Summary
This PR encapsulates the body of work related to a bug with how SSO auto sign-in work when the `loginAttempted` cookie is present (due to users not clearing cookies). 

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#82673

## Testing done
Unit testing + manual

## Screenshots
n/a

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

How to test
0. Start `vets-website` and `vets-api`
1. The following items need to be set in `localStorage` for the browser. Copy and paste the below code into the console of your browser after starting `vets-website`.

```js
localStorage.setItem('loginAttempted', true); // prevents infinite loop of attempting ot call `custom` endpoint
localStorage.setItem('hasSessionSSO', true); // set by the keepalive request
localStorage.setItem('sessionExpirationSSO', new Date().toString()) // set by keepalive request
```

2. In local environments, the `/keepalive` call does not exist and instead a mock call is used by modifying query parameters. `keepalive-ttl`, `keepalive-authn`, and `csp_type` are the query parameters that are used to mock the `/keepalive` header values. 
    - (User should be logged in) http://localhost:3001/?keepalive-ttl=900&keepalive-authn=http://idmanagement.gov/ns/assurance/loa/3&csp_type=IDME
    - (User should no be logged in if TTL is 0) http://localhost:3001/?keepalive-ttl=0&keepalive-authn=http://idmanagement.gov/ns/assurance/loa/3&csp_type=IDME
